### PR TITLE
feat(ui) Port oxfordizeArray from getsentry

### PR DESCRIPTION
**Pull request by markstory**
Bring this utility module to sentry and replace bespoke implementations with the new function. I'll remove the getsentry twin after this merges.

### CODE REVIEW

File: static/app/components/charts/utils.tsx [hunk 1]
```diff
@@ -322,9 +323,8 @@ export const getPreviousSeriesName = (seriesName: string) => {
 
 function formatList(items: Array<string | number | undefined>) {
   const filteredItems = items.filter(item => !!item);
-  return [[...filteredItems].slice(0, -1).join(', '), [...filteredItems].slice(-1)]
-    .filter(type => !!type)
-    .join(' and ');
+
+  return oxfordizeArray(filteredItems.filter(type => !!type));
```
Comment (billyvg) [inline static/app/components/charts/utils.tsx:327]: Isn't `filteredItems` already filtered? in L325?

File: static/app/utils/oxfordizeArray.tsx [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+//
+// for example: ['A'] --> 'A'
+//              ['A', 'B'] --> 'A and B'
+//              ['A', 'B', 'C'] --> 'A, B, and C'
+const oxfordizeArray = (strings: string[]) =>
+  strings.length <= 2
+    ? strings.join(' and ')
+    : [strings.slice(0, -1).join(', '), strings.slice(-1)[0]].join(', and ');
+
+export const oxfordizeElements = (elements: JSX.Element[]): JSX.Element => {
```
Comment (JonasBa) [inline static/app/utils/oxfordizeArray.tsx:14]: Should this be NonNullable<React.ReactNode> to be a bit more permissive? E.g. you could then use this as a mix of oxfordizeElements([num, str, whatever type]).

File: static/app/utils/oxfordizeArray.tsx [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  if (elements.length === 0) {
```
Comment (JonasBa) [inline static/app/utils/oxfordizeArray.tsx:15]: It feels to me like this should just be a React Component with props {children: React.ReactNode} and we can then use React.Children(children).toArray() to perform the oxfordize logic. That way we have a declarative API for React components + it's "functional" counterpart for string[] types. Wdyt?

Comment (markstory) [reply static/app/utils/oxfordizeArray.tsx:15]: Updated to use a component with children.

File: static/app/utils/oxfordizeArray.spec.tsx [hunk 4]
```diff
@@ -18,14 +18,22 @@ describe('oxfordizeArray', function () {
...
     expect(screen.getByText('three')).toBeInTheDocument();
     expect(screen.getByText(/, and/)).toBeInTheDocument();
   });
+
+  it('correctly formats mixed lists of nodes', function () {
+    const items = [<i key="1">one</i>, 'two'];
+    render(<Oxfordize>{items}</Oxfordize>);
+
+    expect(screen.getByText('one')).toBeInTheDocument();
+    expect(screen.getByText(/, and two/)).toBeInTheDocument();
```
Comment (billyvg) [inline static/app/utils/oxfordizeArray.spec.tsx:37]: Isn't this different behavior than the non-element version where it's `A and B`?

**Review by billyvg — APPROVED** [review]

File: static/app/utils/oxfordizeArray.spec.tsx (hunk 4 above)
Comment (markstory) [reply static/app/utils/oxfordizeArray.spec.tsx:37]: Indeed, the implementation in getsentry was wrong, I'll fix that up.

**Pull request merged**